contracts/limiter/GlobalSystemExitRateLimiter.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity =0.8.13;
+
+import {IGSERL} from "./IGSERL.sol";
+import {CoreRefV2} from "../refs/CoreRefV2.sol";
+import {VoltRoles} from "../core/VoltRoles.sol";
+import {RateLimitedV2} from "../utils/RateLimitedV2.sol";
+
+/// @notice contract to control the flow of funds through the system with a rate limit.
+/// In a bank run due to losses exceeding the surplus buffer, this will allow the system
+/// to apply a uniform haircut to all users after the buffer is depleted.
+/// All minting should flow through this smart contract.
+/// Non Custodial Peg Stability Modules will be granted the VOLT_RATE_LIMITED_DEPLETER_ROLE
+/// to deplete the buffer through this contract on a global rate limit.
+/// ERC20Allocator will be granted both the VOLT_RATE_LIMITED_DEPLETER_ROLE and VOLT_RATE_LIMITED_REPLENISH_ROLE
+/// to be able to replenish and deplete the buffer.
+contract GlobalSystemExitRateLimiter is IGSERL, RateLimitedV2 {
+    /// @param _core reference to the core smart contract
+    /// @param _maxRateLimitPerSecond maximum rate limit per second that governance can set
+    /// @param _rateLimitPerSecond starting rate limit per second for Volt minting
+    /// @param _bufferCap cap on buffer size for this rate limited instance
+    constructor(
+        address _core,
+        uint256 _maxRateLimitPerSecond,
+        uint128 _rateLimitPerSecond,
+        uint128 _bufferCap
+    )
+        CoreRefV2(_core)
+        RateLimitedV2(_maxRateLimitPerSecond, _rateLimitPerSecond, _bufferCap)
+    {}
+
+    /// @notice anytime PCV is pulled from a PCV deposit where it can be redeemed,
+    /// or it is being redeemed, call in and deplete this buffer
+    /// Pausable and depletes the global buffer
+    /// @param amount the amount of dollars to deplete the buffer by
+    function depleteBuffer(
+        uint256 amount
+    )
+        external
+        /// checks
+        onlyVoltRole(VoltRoles.VOLT_RATE_LIMITED_DEPLETER_ROLE)
+        /// system must be level 1 locked before this function can execute
+        /// asserts system is inside higher level operation when this function is called
+        globalLock(2)
```

Comment:
idk if a global lock is very useful here

Reply:
there are no external calls in deplete and replenish buffer except to core to validate the sender and their roles. it doesn't cost much to do a warm write to storage, so I don't see any harm in doing it. the only time a user will have to pay this additional gas cost is during a redemption through the non custodial PSM which is a very expensive operation on its own.